src/controller.coffee

```diff
@@ -302,9 +303,11 @@ define ['droplet-helper',
       @redrawMain()
       @rebuildPalette()
 
-      # If we were given an unrecognized mode, flip into text mode
-      unless @mode?
-        @setEditorState false
+      # If we were given an unrecognized mode or asked to start in text mode,
+      # flip into text mode here
+      blockMode = @mode? && !@options.textModeAtStart
+      # Always call @setEditorState to ensure palette is positioned properly
+      @setEditorState blockMode
```

Comment:
I might call this inBlockMode or useBlockMode to make it a little bit more clear that this is a bool, unlike @mode